### ISSUE TITLE
Miscellaneous problems on Jcomments 4.0.26

### ISSUE THREAD

**Issue by philevr** [exstreme/Jcomments-4#123]
After bug correction (inserted in 4.0.26) for my Jcomments migration to Joomla 4, some of my tests brought following miscellaneous problems :

- When sending a new comment for an article, delay for performing validation and display of the comment seems very long with wheel rotation during a long period
- Line break written inside the comments are not taken into account : in the final comment (and in the backend), line feeds br / are not inserted in place of line breaks in the html code
- Icons for refresh, RSS flows and subscribe are not displayed : icon-loop, icon-rss and icon-mail not displaying (I do not know how display of those icons is managed)
- Gravatar image is not displayed though indicated in autorisations settings (Super Admin and Registered)

See attached image of a personal comment (my gravatar still working on Jcomments 3.x).

![Comment-Jcomments 4 0;26](https://user-images.githubusercontent.com/7049368/211574101-021c7f4a-a57e-4de6-99fa-7b3a34941e92.jpg)

Are those problems already reported ?

Thanks.

**Comment by Globulopolis** [comment]
Can you provide an access to your site? A cannot confirm these issues at this time.

**Comment by Globulopolis** [comment]
> - Icons for refresh, RSS flows and subscribe are not displayed : icon-loop, icon-rss and icon-mail not displaying (I do not know how display of those icons is managed)

You template must load file `media/system/css/joomla-fontawesome.min.css`

**Comment by philevr** [comment]
Sorry, you can't access my site : at the moment, it is only local with Wamp and I am testing before my real site migration (https://corse-sauvage.com) !
OK for icons, I'll check to-morrow.

**Comment by philevr** [comment]
For Gravatar : 

- In Jcomments 3.x, Gravatar did not need an avatar plugin and could be displayed directly by jcomments.php
- In Jcomments 4.x, I browsed jcomments.php code and did not find Gravatar code used in 3.x

**Comment by Globulopolis** [comment]
> at the moment, it is only local with Wamp and I am testing before my real site migration

In this situation I would recommend to enable logging in Joomla and try to find anything in PHP and Joomla logs.


> For Gravatar :
> 
>     * In Jcomments 3.x, Gravatar did not need an avatar plugin and could be displayed directly by jcomments.php
> 
>     * In Jcomments 4.x, I browsed jcomments.php code and did not find Gravatar code used in 3.x

In the new version, a duplicate Gravatar code from the component is removed. Use avatar plugin from here https://github.com/exstreme/Jcomments-4/blob/master/build/plugins/README.md

**Comment by philevr** [comment]
OK for Gravatar with the new plugin and for icons display !
Still have problem with long delay for sending a comment and line break...

Thanks for your help !

**Comment by exstreme** [comment]
Turn on debug, you can see, what is slow by senfing  query

**Comment by philevr** [comment]
I don't know why, but slow comment sending processing disappeared since this morning !
I think there is a problem of wiki in my Jcomments 4 version : no line break and list process.
You can see diffrences between my Jcomments 3 and 4 versions in the attached images with the two html creations.
![Sans titre - 9](https://user-images.githubusercontent.com/7049368/212072209-d03d282b-dd84-46f2-a648-13a1fe8b20b1.jpg)
![Sans titre - 10](https://user-images.githubusercontent.com/7049368/212072243-a7795e4c-9cdf-4c9c-b6d0-41a9caf08278.jpg)
![Sans titre - 7](https://user-images.githubusercontent.com/7049368/212072286-91c7bbed-b06f-4a03-a20a-2c36608b8fe6.jpg)

**Comment by Globulopolis** [comment]
Open file `administrator/components/com_jcomments/models/comment.php`, go to line №125 and replace:
`//$table->comment = JCommentsText::nl2br($table->comment); // TODO Remove JCommentsText::nl2br()`
 with
`$table->comment = JCommentsText::nl2br($table->comment);`

Save file.

Let me know if would help.

**Comment by philevr** [comment]
No change while trying my last comment !
![Sans titre - 11](https://user-images.githubusercontent.com/7049368/212124518-6aae5503-c76a-4986-b7cd-b6934d837834.jpg)

**Comment by philevr** [comment]
And, as you can see, wiki "list" is not processed in adequate manner !

**Comment by Globulopolis** [comment]
> And, as you can see, wiki "list" is not processed in adequate manner while bold, italic, ..., img and link are OK. Another difference with Jcomments 3 : [url=link adress]something[/url] seems not working !

Try to edit comment and place a new lines again.

**Comment by philevr** [comment]
OK : I edited my last comment as SuperAdmin and added 3 new lines : they stay inline (see attached image).
![Sans titre - 12](https://user-images.githubusercontent.com/7049368/212137168-679eea47-f2e8-497b-8844-c80f8c241e4a.jpg)

**Comment by Globulopolis** [comment]
One last thing...
Open file `administrator/components/com_jcomments/tables/comment.php` go to line №327 and change
`//$value = JCommentsText::nl2br($value);`
with
`$value = JCommentsText::nl2br($value);`

Save file. And re-save comment.

**Comment by philevr** [comment]
Same as previous test.
After having modified administrator/components/com_jcomments/tables/comment.php, I tried a new comment with same wiki as before and after display of this new comment I tried to edit it by adding some lines. All lines break are ignored and all wiki elements are inline...

**Comment by Globulopolis** [comment]
> Still have problem with long delay for sending a comment

Is the sending of notifications(on email) enabled? If yes, this will result to huge delays while comment saving due to delays when sending mail.

**Comment by philevr** [comment]
Sorry with that late answer, but I was busy with other subjects !
In my configuration, notifications and report to administrator are disabled. When I send a comment, I just have the box "to be notified with following comments", but I never check that box.
I have still some delay with each comment sent, but shorter than at the beginning and only a few seconds maximum.

Just for your information : 
-I have just migrated another site with Jcomments and encountered a problem I did not expect. with impossibility of sending any comment
- I thought first that it could be another issue to open, but I saw a message on Joomla administrator panel reporting incorrect jcomments tables in mySQL data base
- After repair of those tables, sending comment  was again possible.

Perhaps it would be interesting to notify Jcomments 4 users with the possibility that Jcomments tables could be impacted during Joomla migration and that it would be necessary to check them after the process.
